pyntc/devices/ios_device.py

```diff
@@ -685,6 +685,80 @@ def file_copy_remote_exists(self, src, dest=None, file_system=None):
         log.debug("Host %s: File %s does not already exist on remote.", self.host, src)
         return False
 
+    def file_download(
+        self, protocol, src, path, username=None, password=None, dest="", file_system=None, read_timeout=2000
```

Comment:
To me... we should just provide the url, it will be different for many different use cases, and the user can figure out how they want to handle usernames or not.